lib/app/sequence_base.rb

```diff
@@ -808,7 +808,7 @@ def get_value_for_search_param(element, include_system = false)
                        when FHIR::CodeableConcept
                          if include_system
                            coding_with_code = resolve_element_from_path(element, 'coding') { |coding| coding.code.present? }
-                           "#{coding_with_code.system}|#{coding_with_code.code}"
+                           coding_with_code.present? ? "#{coding_with_code.system}|#{coding_with_code.code}" : nil
```

Comment:
Is this nil-safe everywhere? As an example, I see nil checks here: https://github.com/onc-healthit/inferno-program/blob/development/lib/modules/uscore_v3.1.0/us_core_careplan_sequence.rb#L222
but not here: https://github.com/onc-healthit/inferno-program/blob/development/lib/modules/uscore_v3.1.0/us_core_careplan_sequence.rb#L184

Reply:
We don't explicitly check for nil in the second example, but the validate_reply_entries acts as a nil check since we won't reach that line if there was no category in the resource.